artemis-web/src/main/java/org/apache/activemq/artemis/component/WebServerComponent.java

```diff
@@ -145,9 +146,16 @@ public void internalStop() throws Exception {
                //tmpdir will be removed by deleteOnExit()
                //somehow when broker is stopped and restarted quickly
                //this tmpdir won't get deleted sometimes
-               boolean fileDeleted = TimeUtils.waitOnBoolean(false, 10000, tmpdir::exists);
+               boolean fileDeleted = TimeUtils.waitOnBoolean(false, 5000, tmpdir::exists);
                if (!fileDeleted) {
-                  ActiveMQWebLogger.LOGGER.tmpFileNotDeleted(tmpdir);
+                  //because the execution order of shutdown hooks are
+                  //not determined, so it's possible that the deleteOnExit
+                  //is executed after this hook, in that case we force a delete.
+                  FileUtil.deleteDirectory(tmpdir);
+                  ActiveMQWebLogger.LOGGER.tmpFileForceDelete(tmpdir);
```

Comment:
ActiveMQWebLogger.LOGGER.debug("..."); would also work... but I prefer to have the actual logger on debug or trace